client/components/Search.jsx

```diff
@@ -26,8 +25,8 @@ export default class Search extends PureComponent {
             value={query}
             placeholder="Enter regexp"
             onInput={this.handleValueChange}
+            onKeyDown={this.handleKeyDown}
             onBlur={this.handleInputBlur}/>
-          <Button className={s.clear} onClick={this.handleClearClick}>x</Button>
```

Comment:
I actually liked the clear search button. Could we have both key handler and the button?

Reply:
I've removed it because of this "side effect":

![1](https://user-images.githubusercontent.com/302213/44835428-d10b6400-ac3c-11e8-9da9-6b2cb7eae836.gif)

But you're right - it's better to return it back and think how to solve this issue.